hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconPipelineReportHandler.java

```diff
@@ -63,14 +64,28 @@ protected void processPipelineReport(PipelineReport report,
     PipelineID pipelineID = PipelineID.getFromProtobuf(report.getPipelineID());
     if (!reconPipelineManager.containsPipeline(pipelineID)) {
       LOG.info("Unknown pipeline {}. Trying to get from SCM.", pipelineID);
-      Pipeline pipelineFromScm =
-          scmServiceProvider.getPipeline(report.getPipelineID());
+      Pipeline pipelineFromScm;
+
+      try {
+        pipelineFromScm =
+            scmServiceProvider.getPipeline(report.getPipelineID());
+      } catch (IOException ex) {
+        if (ex instanceof RemoteException) {
+          IOException ioe = ((RemoteException) ex)
+                  .unwrapRemoteException(PipelineNotFoundException.class);
+          if (ioe instanceof PipelineNotFoundException) {
```

Comment:
In this case do we need to throw PipelineNotFoundException??

Reply:
And also avoid loggig??

catch(NotLeaderException ex) {
        // Avoid NotLeaderException logging which happens when processing
        // pipeline report on followers.
      } catch(PipelineNotFoundException e) {
      // When pipeline not found donot log??
      }catch(IOException e) {
        LOGGER.error("Could not process pipeline report={} from dn={}.",
            report, dn, e);
      }

Reply:
thank for the review , i have updated the PR , please take a look